src/c++/perf_analyzer/genai-perf/tests/test_synthetic_image_generator.py

```diff
@@ -0,0 +1,106 @@
+import base64
+from io import BytesIO
+from pathlib import Path
+from unittest.mock import patch
+
+import numpy as np
+import pytest
+from genai_perf.exceptions import GenAIPerfException
+from genai_perf.llm_inputs.synthetic_image_generator import (
+    ImageFormat,
+    RandomFormatBase64Encoder,
+    SyntheticImageGenerator,
+    images_from_file_generator,
+    white_images_generator,
+)
+from PIL import Image
+
+
+@pytest.mark.parametrize(
+    "image_size",
+    [
+        (100, 100),
+        (200, 200),
+    ],
+)
+def test_different_image_size(image_size):
+    sut = SyntheticImageGenerator(
+        mean_size=image_size,
+        dimensions_stddev=[0, 0],
+        image_iterator=white_images_generator(),
+    )
+
+    image = next(sut)
+
+    assert isinstance(image, Image.Image), "generator produces unexpected type of data"
+    assert image.size == image_size, "image not resized to the target size"
+
+
+def test_negative_size_is_not_selected():
+    sut = SyntheticImageGenerator(
+        mean_size=(-1, -1),
+        dimensions_stddev=[10, 10],
+        image_iterator=white_images_generator(),
+    )
+
+    # exception is raised, when PIL.Image.resize is called with negative values
+    next(sut)
+
+
+@patch("pathlib.Path.exists", return_value=False)
+def test_images_from_file_raises_when_file_not_found(mock_exists):
+    DUMMY_PATH = Path("dummy-image.png")
+    sut = images_from_file_generator(DUMMY_PATH)
+
+    with pytest.raises(GenAIPerfException):
+        next(sut)
+
+
+DUMMY_IMAGE = Image.new("RGB", (100, 100), color="blue")
+
+
+@patch("pathlib.Path.exists", return_value=True)
+@patch(
+    "PIL.Image.open",
+    return_value=DUMMY_IMAGE,
```

Comment:
```suggestion
    return_value=Image.new("RGB", (100, 100), color="blue"),
```

Reply:
I'm testing against the `DUMMY_IMAGE` in an assertion below, so I prefer to keep it named, but I moved the variable definition from the global to the local scope.